tools/restatectl/src/commands/log/seal_extend_chain.rs

```diff
@@ -0,0 +1,140 @@
+
+    /// Replication property
+    #[clap(long, required_if_eq("provider", "replicated"))]
+    replication_factor: Option<NonZeroU8>,
+    /// A comma-separated list of the nodes in the nodeset. e.g. N1,N2,N4 or 1,2,3
+    #[clap(long, required_if_eq("provider", "replicated"), value_delimiter=',', num_args = 1..)]
+    nodeset: Vec<PlainNodeId>,
+    /// The generational node id of the sequencer node, e.g. N1:1
+    #[clap(long, required_if_eq("provider", "replicated"))]
+    sequencer: Option<GenerationalNodeId>,
+}
+
+async fn seal_and_extend_chain(
+    connection: &ConnectionInfo,
+    opts: &SealAndExtendChainOpts,
+) -> anyhow::Result<()> {
+    let channel = grpc_connect(connection.cluster_controller.clone())
+        .await
+        .with_context(|| {
+            format!(
+                "cannot connect to cluster controller at {}",
+                connection.cluster_controller
+            )
+        })?;
+
+    let mut client =
+        ClusterCtrlSvcClient::new(channel).accept_compressed(CompressionEncoding::Gzip);
+
+    let (provider, params) = match opts.provider {
+        ProviderKind::Local => (ProtoProviderKind::Local, rand::random::<u64>().to_string()),
+        #[cfg(any(test, feature = "memory-loglet"))]
+        ProviderKind::InMemory => (ProtoProviderKind::InMemory, String::default()),
+        #[cfg(feature = "replicated-loglet")]
+        ProviderKind::Replicated => (
+            ProtoProviderKind::Replicated,
+            replicated_loglet_params(&mut client, opts).await?,
+        ),
+        #[allow(unreachable_patterns)]
+        _ => anyhow::bail!("Provider kind is not supported"),
+    };
+
+    client
+        .seal_and_extend_chain(SealAndExtendChainRequest {
+            log_id: opts.log_id,
+            min_version: Some(Version {
+                value: opts.min_version.get(),
+            }),
+            provider: provider as i32,
+            segment_index: opts.segment_index,
+            params,
+        })
+        .await?;
+
+    Ok(())
+}
+
+async fn replicated_loglet_params(
+    client: &mut ClusterCtrlSvcClient<Channel>,
+    opts: &SealAndExtendChainOpts,
+) -> anyhow::Result<String> {
+    let tail_index = match opts.segment_index {
+        Some(index) => index,
+        None => {
+            let mut logs_resposne = client
+                .list_logs(ListLogsRequest {})
+                .await
+                .context("failed to get logs chain")?
+                .into_inner();
+
+            let logs = StorageCodec::decode::<Logs, _>(&mut logs_resposne.logs)?;
+            let chain = logs
+                .chain(&LogId::from(opts.log_id))
+                .with_context(|| format!("Unknown log id '{}'", opts.log_id))?;
+
+            chain.tail_index().into()
+        }
+    };
+
+    // format is, log_id in the higher order u32, and segment_index in the lower
+    let loglet_id: u64 =
+        (u64::from(opts.log_id) << (size_of::<LogId>() * 8)) + u64::from(tail_index + 1);
```

Comment:
@tillrohrmann this need to be aligned with whatever we do in cluster controller. Should we have a common utility in ReplicatedLogletId to generate them?